File: sdk/python/aiinfra/gateway.py
from typing import Any

from aiinfra._http import HTTPClient


class GatewayClient:
    def __init__(
        self,
        base_url: str = "http://localhost:8080",
        *,
        admin_token: str | None = None,
        api_key: str | None = None,
        timeout: float = 120.0,
    ):
        self._http = HTTPClient(
            base_url,
            admin_token=admin_token,
            api_key=api_key,
            timeout=timeout,
        )

    def close(self) -> None:
        self._http.close()

    def __enter__(self):
        return self

    def __exit__(self, *args):
        self.close()

    def health(self) -> dict:
        return self._http.get("/health")

    def ready(self) -> dict:
        return self._http.get("/ready")

    def list_models(self) -> dict:
        return self._http.get("/admin/models")

    def list_api_keys(self) -> list[dict]:
        return self._http.get("/admin/keys")

    def create_api_key(
        self,
        name: str,
        *,
        tenant_id: str = "default",
        rate_limit_rpm: int | None = None,
    ) -> dict:
        payload: dict[str, Any] = {"name": name, "tenant_id": tenant_id}
        if rate_limit_rpm is not None:
            payload["rate_limit_rpm"] = rate_limit_rpm
        return self._http.post("/admin/keys", json=payload)

    def chat_completions(
        self,
        model: str,
        messages: list[dict[str, str]],
        *,
        stream: bool = False,
        **extra: Any,
    ) -> dict:
        payload = {"model": model, "messages": messages, "stream": stream, **extra}
        return self._http.post("/v1/chat/completions", json=payload)

File: sdk/typescript/src/gateway.ts
import { HttpClient } from './http.js';

export class GatewayClient {
  private http: HttpClient;

  constructor(baseUrl = 'http://localhost:8080', opts: { adminToken?: string; apiKey?: string } = {}) {
    this.http = new HttpClient(baseUrl, opts);
  }

  health() {
    return this.http.get('/health');
  }

  listModels() {
    return this.http.get<{ data: { id: string; provider: string }[] }>('/admin/models');
  }

  listApiKeys(tenantId?: string) {
    const q = tenantId ? `?tenant_id=${encodeURIComponent(tenantId)}` : '';
    return this.http.get(`/admin/keys${q}`);
  }

  createApiKey(name: string, tenantId = 'default', rateLimitRpm?: number) {
    return this.http.post('/admin/keys', { name, tenant_id: tenantId, rate_limit_rpm: rateLimitRpm });
  }

  chatCompletions(model: string, messages: { role: string; content: string }[], extra: Record<string, unknown> = {}) {
    return this.http.post('/v1/chat/completions', { model, messages, stream: false, ...extra });
  }

  listTenants() {
    return this.http.get('/admin/tenants');
  }

  tenantUsage(tenantId: string) {
    return this.http.get(`/admin/tenants/${tenantId}/usage`);
  }
}
